R/djpr_save_pptx.R

```diff
@@ -0,0 +1,102 @@
+# This file implements the `djpr_save_pptx` function which takes a ggplot
+# and exports it as a PowerPoint slide.
+
+#' Export a ggplot to a DJPR SPP Powerpoint slide.
+#'
+#' The exported slide will use the SPP template. All text is editable on the
+#' slide. Users can choose from several slide formats.
+#'
+#' @param destination File path to save the slide to, including extension,
+#' such as "output.pptx".
+#' @param plot The ggplot to turn into a slide.
+#' Default is `ggplot2::last_plot()`.
+#' @param layout Slide layout to use. Options are "full", "half", and
+#' "twothirds". The layouts refer to the proportion of the slide width taken
+#' up by the chart.
+#' @param signpost Signpost heading, if required. If `NULL` (the default), no
+#' signpost will be included.
+#'
+#' @return Since the function saves the slide to a file, the return value
+#' is not defined and may change in future
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' library(ggplot2)
+#' # First, create a ggplot2 object
+#' the_ggplot <- ggplot(mtcars, aes(x = wt, y = mpg)) +
+#'   geom_point() +
+#'   labs(title = "Chart title",
+#'        subtitle = "Chart subtitle",
+#'        caption = "Notes and source information.")
+#'
+#' # Export `the_ggplot` to file `output.pptx`
+#' djpr_save_pptx("output.pptx", the_ggplot)
+#'
+#' # Specify layout and signpost
+#' djpr_save_pptx("output.pptx", the_ggplot, layout = "half", signpost = "section")
+#' }
+djpr_save_pptx <- function(destination,
+                           plot = ggplot2::last_plot(),
```

Comment:
@duyjpr I agree that plot feels more natural as the first argument, but I would rather this ordering, for two reasons:

1. This is consistent with `ggplot2::ggsave()` and I value that consistency; and
2. This ordering enables the use of `ggplot2::last_plot()` as the default value to plot, so users can use the function by specifying only a filename if they like.